### PR TITLE
reader.js refactor and testing

### PR DESCRIPTION
**Pull request by DarkMorford**
* Refactored reader.js to use common peek and read functions, which also throw an exception (from readUInt8) if reading beyond the end of the Buffer.
* Added tests for reader.js functions using Jest framework.
* Added/updated config and editor support files.

### CODE REVIEW

**Comment by hedgehog1029** [comment]
Looks good! One note is that I think we might need to ask all contributors before we relicense (since by default, the license is All Rights Reserved or [No License](https://choosealicense.com/no-permission/)) - I'm fine with MPL-2.0, but we should ask @mrphlip and rf232.

[Edit: I forgot I specified it in package.json before]

**Comment by DarkMorford** [comment]
Yeah, MPL-2.0 was already in the package.json file, I just added the full text.

That's a good point, though; we should probably make sure everyone agrees on the license before we go much further.

**Comment by mrphlip** [comment]
Fine with me

**Comment by rf232** [comment]
Not the biggest fan of the Compatibility with GPL part of MPL, GPLs "or any later version". 
Also we got code samples from the developer, and they didn't contain a license so i don't even know if we are allowed to redistribute in any way.

**Comment by DarkMorford** [comment]
That's actually a very good point. I'll go ahead and explicitly mark it as unlicensed for now, and we can always re-license it later if we decide we can and want/need to.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # gloomparser
 
-Parsing Gloomhaven Helper's state file for external tracking
+Parsing Gloomhaven Helper's state file for external tracking.
 
-Originally written by rf232 and refactored a little by @hedgehog1029
+Originally written by rf232 and refactored a little by @hedgehog1029 and @DarkMorford.
```
Comment (rf232) [inline README.md]:
```suggestion
Originally written by @rf232 and refactored a little by @hedgehog1029 and @DarkMorford.
```

File: package.json [hunk 2]
```diff
@@ -24,7 +25,8 @@
     "rf232",
```
Comment (rf232) [inline package.json]:
```suggestion
    "rf232 <rob@rob-franken.nl>",
    "hedgehog1029",
```

**Review by rf232 — APPROVED** [review]

**Pull request merged**